test/gst-vaapi/vpp/brightness.py

```diff
@@ -6,45 +6,52 @@
 
 from ....lib import *
 from ..util import *
+from .vpp import VppTest
 
 spec = load_test_spec("vpp", "brightness")
-
-@slash.requires(have_gst)
-@slash.requires(*have_gst_element("vaapi"))
-@slash.requires(*have_gst_element("vaapipostproc"))
-@slash.requires(*have_gst_element("checksumsink2"))
-@slash.parametrize(*gen_vpp_brightness_parameters(spec))
-@platform_tags(VPP_PLATFORMS)
-def test_default(case, level):
-  params = spec[case].copy()
-  params.update(
-    level = level, mlevel = mapRange(level, [0, 100], [-1.0, 1.0]),
-    mformat = mapformat(params["format"]),
-    hwup_format = mapformat_hwup(params["format"]))
-  params["ofile"] = get_media()._test_artifact(
-    "{}_brightness_{level}_{format}_{width}x{height}"
-    ".yuv".format(case, **params))
-
-  call(
-    "gst-launch-1.0 -vf filesrc location={source} num-buffers={frames}"
-    " ! rawvideoparse format={mformat} width={width} height={height}"
-    " ! videoconvert ! video/x-raw, format={hwup_format}"
-    " ! vaapipostproc format={mformat} width={width} height={height}"
-    " brightness={mlevel} ! checksumsink2 file-checksum=false"
-    " frame-checksum=false plane-checksum=false dump-output=true"
-    " dump-location={ofile}".format(**params))
-
-  psnr = calculate_psnr(
-    params["source"], params["ofile"],
-    params["width"], params["height"],
-    params["frames"], params["format"])
-
-  assert psnr[-2] == 100, "Cb(U) should not be affected by BRIGHTNESS filter"
-  assert psnr[-1] == 100, "Cr(V) should not be affected by BRIGHTNESS filter"
-
-  def compare(k, ref, actual):
-    assert ref is not None, "Invalid reference value"
-    assert abs(ref[-3] - actual[-3]) <  0.2, "Luma (Y) out of baseline range"
-
-  get_media().baseline.check_result(
-    compare = compare, context = params.get("refctx", []), psnr = psnr)
+spec_r2r = load_test_spec("vpp", "brightness", "r2r")
+
+class Brightness(VppTest):
+  def before(self):
+    vars(self).update(
+      vpp_element = "brightness"
+    )
+    super(Brightness, self).before()
+
+  def init(self, tspec, case, level):
+    vars(self).update(tspec[case].copy())
+    vars(self).update(
+      level = level, mlevel = mapRange(level, [0, 100], [-1.0, 1.0]),
+      mformat = mapformat(self.format),
+      hwup_format = mapformat_hwup(self.format))
+
+  @slash.parametrize(*gen_vpp_brightness_parameters(spec))
+  @platform_tags(VPP_PLATFORMS)
+  def test_default(self, case, level):
```

Comment:
If we use `default` as the class name, then `def test(...)` is good enough here.  For generated test names, this would produce something less redundant, e.g.:
```
test/gst-vaapi/vpp/brightness.py:default.test(case=QCIF,level=0)
```